### PR TITLE
Replace treeview with material-ui-treeview

### LINKED ISSUE #2749: Remove material-ui-treeview
We started using https://github.com/helfi92/material-ui-treeview/ in https://community-tc.services.mozilla.com/hooks because material-ui did not have a tree view component available at the time, but now they do. Let's move away from what we were using and use material-ui's [TreeView](https://material-ui.com/api/tree-view/#treeview-api) component. The [demo](https://material-ui.com/components/tree-view/#rich-object) shows that they support rendering the same tree structure that we have so switching to it shouldn't be too demanding from our end.

### PR DESCRIPTION
**Pull request by rohanharikr**
Github Bug/Issue: Fixes #2749

### CODE REVIEW

**Comment by rohanharikr** [comment]
The code is not working as intended. This is what I tried .... I'm running out of ideas. Any idea what's happening?

This block seems to be causing this issue, althought not fully sure.

    const renderTree = (nodes) => (nodes.map(
      <TreeItem key={count++} nodeId={count} label={nodes.hookGroupId}>
       {Array.isArray(nodes.hooks) ? nodes.hooks.map((node) => renderTree(node)) : null}
      </TreeItem>
    ));

Screenshot 2020-05-14 at 11 48 05 AM

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 1]
```diff
@@ -51,19 +54,16 @@ export default class ListHooks extends Component {
...
-              hook.hookId
-            )}`,
-          })),
-        }))
-      : [];
+
+    let count = 0;
+
+    const renderTree = (nodes) => (nodes.map(
+      <TreeItem key={count++} nodeId={count} label={nodes.hookGroupId}>
```
Comment (helfi92) [inline ui/src/views/Hooks/ListHooks/index.jsx:63]: `nodes.map` [takes a callback function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) as the first argument but is currently being fed a React element. That probably explains why it's acting weird. Hope that helps.

**Comment by helfi92** [comment]
👋 How is it going @rohanharikr? :-)

**Comment by rohanharikr** [comment]
> 👋 How is it going @rohanharikr? :-)

Hello @helfi92  ! I didn't work on it after the last commit since I was not able to figure it out but I have been reading a lot and watching stuff, so I believe that I am a bit better/understand react a little bit better now. I will give it a go again this weekend (by 14th june eod) and update you on the status.

**Comment by helfi92** [comment]
Sounds good and no problem. Thank you for the update!

**Comment by djmitche** [comment]
@rohanharikr would you like another round of review?  If so, click the 
image
in the upper-right corner of the PR page.

**Comment by rohanharikr** [comment]
> @rohanharikr would you like another round of review? If so, click the
> image
> in the upper-right corner of the PR page.

Hello @djmitche 

Thanks for this. I need to make a couple of changes before I put it up for review. Hopefully, should be done by Monday.

Cheers

**Comment by rohanharikr** [comment]
Had difficulty and needed help with following:
1. Links not working
2. Styling like padding between items
3. A loop to render in jsx (right now, it is just hard coded)

Thank you so much in advance. :)

**Comment by helfi92** [comment]
> Had difficulty and needed help with following:
> 
>     1. Links not working

```diff
diff --git a/ui/src/views/Hooks/ListHooks/index.jsx b/ui/src/views/Hooks/ListHooks/index.jsx
index 1bee631d6..71cb53033 100644
--- a/ui/src/views/Hooks/ListHooks/index.jsx
+++ b/ui/src/views/Hooks/ListHooks/index.jsx
@@ -14,6 +14,7 @@ import HelpView from '../../../components/HelpView';
 import Search from '../../../components/Search';
 import Button from '../../../components/Button';
 import ErrorPanel from '../../../components/ErrorPanel';
+import Link from '../../../utils/Link';
 import hooksQuery from './hooks.graphql';
 
 @hot(module)
@@ -66,16 +67,24 @@ export default class ListHooks extends Component {
           })),
         }))
       : [];
-    const renderTree = treeItem => (
-      <TreeItem
-        key={treeItem.value}
-        nodeId={treeItem.value}
-        label={treeItem.value}>
-        {Array.isArray(treeItem.nodes)
-          ? treeItem.nodes.map(node => renderTree(node))
-          : null}
-      </TreeItem>
-    );
+    const renderTree = treeItem => {
+      const item = (
+        <TreeItem
+          key={treeItem.value}
+          nodeId={treeItem.value}
+          label={treeItem.value}>
+          {Array.isArray(treeItem.nodes)
+            ? treeItem.nodes.map(node => renderTree(node))
+            : null}
+        </TreeItem>
+      );
+
+      if (treeItem.href) {
+        return <Link to={treeItem.href}>{item}</Link>;
+      }
+
+      return item;
+    };

```


> 
>     2. Styling like padding between items

If you go to https://material-ui.com/components/tree-view/ and click on "Show Source" on any of the examples there you should be able to see how styling is applied.

> 
>     3. A loop to render in jsx (right now, it is just hard coded)

```diff
diff --git a/ui/src/views/Hooks/ListHooks/index.jsx b/ui/src/views/Hooks/ListHooks/index.jsx
index 1bee631d6..d02f6de8f 100644
--- a/ui/src/views/Hooks/ListHooks/index.jsx
+++ b/ui/src/views/Hooks/ListHooks/index.jsx
@@ -95,11 +95,7 @@ export default class ListHooks extends Component {
             defaultCollapseIcon={<ExpandMoreIcon />}
             defaultExpanded={['root']}
             defaultExpandIcon={<ChevronRightIcon />}>
-            {renderTree(tree[0])}
-            {renderTree(tree[1])}
-            {renderTree(tree[2])}
-            {renderTree(tree[3])}
-            {renderTree(tree[4])}
+            {tree.map(renderTree)}
           </TreeView>
         )}
         <Button
```


> 
> Thank you so much in advance. :)

Hope this helps.

**Comment by djmitche** [comment]
I'm going to close this as un-touched for a while.  If you'd like to continue working on it, just make a comment and we can re-open it!

**Pull request closed without merging**